server/src/main/java/org/opensearch/index/store/RemoteDirectory.java

```diff
@@ -340,7 +350,7 @@ private void uploadBlob(
             WritePriority.NORMAL,
             (size, position) -> uploadRateLimiter.apply(new OffsetRangeIndexInputStream(from.openInput(src, ioContext), size, position)),
             expectedChecksum,
-            this.getBlobContainer() instanceof VerifyingMultiStreamBlobContainer
+            remoteIntegrityEnabled
```

Comment:
We are overriding remote integrity check here because we can't provide the checksum of encrypted content as we only have checksum of raw content. It has nothing to do with support of remote store.